### PR TITLE
Bug Fixes and Improvements to the Shader Pack Option GUI

### PR DESCRIPTION
**Pull request by FoundationGames**
### This pull request addresses some feedback regarding the Shader Pack Option GUI.
image
- Right clicking sliders no longer causes them to step backwards like non-slider options
- The Escape key can be used to navigate back in the option menu, as well as back out of the option menu into the pack selection menu
- Shader pack configuration files can be exported and imported in and out of the option menu
- Configuration files can be dragged into the option menu to import them
- Spaghetti code used for various general-purpose buttons throughout the GUI has been unified
- "Current Profile" message details the number of options the user has changed in *addition* to the profile rather than the overall number of changed options
- "Current Profile" message no longer translates the name of the profile

### CODE REVIEW

**Review by IMS212 — APPROVED** [review]
Looks good to me!

File: src/main/java/net/coderbot/iris/gui/element/widget/SliderElementWidget.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public boolean mouseClicked(double mx, double my, int button) {
 
 			return true;
 		}
-		return super.mouseClicked(mx, my, button);
+		return false;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/gui/element/widget/SliderElementWidget.java:106]: Suggestion: Add a comment here saying why this shouldn't defer to the parent - just for anyone reading the code in the future.

**Review by coderbot16 — CHANGES_REQUESTED** [review]

File: src/main/java/net/coderbot/iris/gui/element/ShaderPackOptionList.java [hunk 2]
```diff
@@ -139,66 +175,56 @@ public void render(PoseStack poseStack, int index, int y, int x, int entryWidth,
...
 			GuiUtil.playButtonClickSound();
 
 			return true;
 		}
 
-		private boolean resetButtonClicked(double mouseX, double mouseY, int button) {
+		private boolean resetButtonClicked(IrisElementRow.TextButtonElement button) {
 			if (Screen.hasShiftDown()) {
 				Iris.resetShaderPackOptionsOnNextReload();
 				this.screen.applyChanges();
```
[the remark this answers is not in the document]
Comment (FoundationGames) [reply src/main/java/net/coderbot/iris/gui/element/ShaderPackOptionList.java:230]: The GUI has jobs that need to be handled on reload, therefore it's required that reloading happen through the GUI (I may add a comment explaining this)

**Review by coderbot16 — APPROVED** [review]
Great work, thanks! The import / export feature is certainly a welcome addition.

**Pull request merged**